ruma-api/src/lib.rs

```diff
@@ -237,6 +234,16 @@ pub trait Endpoint:
 
     /// Metadata about the endpoint.
     const METADATA: Metadata;
+
+    /// Tries to convert this request into an `http::Request`.
+    ///
+    /// This method should only fail when called on endpoints that require authentication. It may
+    /// also fail with a serialization error in case of bugs in Ruma though.
+    fn try_into_http_request(
+        self,
+        homeserver: &str,
```

Comment:
`homeserver_url` or `homeserver_base_url` (in reference to `HomeserverInfo` / `.well-known` autodiscovery) might be more distinct from `server_name` as in identifiers, so that API users remember this is not just the server domain as in mxids but an actual url that might be totally distinct from the `server_name`.